### DISCUSSION TITLE
Having difficulty with vite_rails and System Tests: "can't find entrypoints/application.js ..."

### DISCUSSION THREAD

**Discussion by radavis** [ElMassimo/vite_ruby#361] [Q&A]
So, first off, thanks for this awesome tool!

I'm attempting to do some system testing and I'm having some difficulty. I've run `bin/rake assets:precompile` and disabled autoBuild with the environment variable `VITE_RUBY_AUTO_BUILD=false`.

From what I can gather, the file exists (`entrypoints/application.js` points to file: `assets/application-d0aa1163.js`). 

Any help or direction here would be greatly appreciated!

**Details**

Error message [[ci-output](https://gitlab.com/radavis/cbt-backend/-/jobs/4205951598)]:

```bash
Vite Ruby can't find entrypoints/application.js in public/vite-test/manifest.json or public/vite-test/manifest-assets.json
```

Contents of `public/vite-test/manifest.json`:

```json
{
  "_index-03be2d59.js": {
    "file": "assets/index-03be2d59.js"
  },
  "entrypoints/application.js": {
    "css": [
      "assets/App-17b4a47d.css"
    ],
    "file": "assets/application-d0aa116e.js",
    "imports": [
      "entrypoints/components/index.jsx",
      "_index-03be2d59.js",
      "entrypoints/components/App.jsx",
      "entrypoints/components/fields/index.js",
      "entrypoints/components/fields/RangeInput.jsx",
      "entrypoints/components/fields/TextAreaInput.jsx",
      "entrypoints/components/fields/TextInput.jsx",
      "entrypoints/components/fields/YesNoInput.jsx",
      "entrypoints/components/SessionComplete.jsx",
      "entrypoints/components/Quote.jsx"
    ],
    "isEntry": true,
    "src": "entrypoints/application.js"
  },
  // ...
}
```

File listing:

```bash
d1a5034e577e:/home/app# ls -l public/vite-test/assets
total 192
-rw-r--r--    1 root     root            38 May  1 14:06 App-17b4a47d.css
-rw-r--r--    1 root     root         10251 May  1 14:06 App-f3a7e735.js
-rw-r--r--    1 root     root           190 May  1 14:06 Quote-a58f81e4.js
-rw-r--r--    1 root     root           311 May  1 14:06 RangeInput-53d41085.js
-rw-r--r--    1 root     root           395 May  1 14:06 SessionComplete-664aa7f6.js
-rw-r--r--    1 root     root           267 May  1 14:06 TextAreaInput-9fdd3fcc.js
-rw-r--r--    1 root     root           276 May  1 14:06 TextInput-9d769e59.js
-rw-r--r--    1 root     root           399 May  1 14:06 YesNoInput-986d455a.js
-rw-r--r--    1 root     root           337 May  1 14:06 application-d0aa116e.js
-rw-r--r--    1 root     root          6826 May  1 14:06 index-03be2d59.js
-rw-r--r--    1 root     root           379 May  1 14:06 index-24e59ef3.js
-rw-r--r--    1 root     root        134490 May  1 14:06 index-ca2077ad.js
-rw-r--r--    1 root     root          1571 May  1 14:06 steps-069ad5f1.json
```

**Comment by ElMassimo** [comment]
Hi Richard!

From the [guide](https://vite-ruby.netlify.app/guide/development.html#integration-tests-in-the-ci-%E2%9C%85):

> it's recommended to run `bin/rake assets:precompile` in a previous CI step.

`autoBuild` is [correctly disabled](https://gitlab.com/radavis/cbt-backend/-/blob/ci/.gitlab-ci.yml#L109) in your CI config, but I don't see [`bin/rake assets:precompile` in your CI pipeline](https://gitlab.com/radavis/cbt-backend/-/blob/ci/.gitlab-ci.yml#L117).

**Reply by radavis** [reply]
Ah, I'm doing that in my [Dockerfile](https://gitlab.com/radavis/cbt-backend/-/merge_requests/8/diffs#6651ddff6eb82c840ced7c1dddee15c6e1913dd4_0_43)

**Reply by ElMassimo** [reply]
The CI errors are not [printing the manifest contents](https://github.com/ElMassimo/vite_ruby/blob/main/vite_ruby/lib/vite_ruby/missing_entrypoint_error.rb#L19), so you might want to add similar code to verify if the manifest contains the expected entrypoints.

You could add the following line before `vite_javascript_tag` in your layout:

```ruby
<%= Kernel.puts JSON.pretty_generate(vite_manifest.refresh) %>
```

**Comment by radavis** [comment]
As it turns out, I had an issue with my Dockerfile and the order I was running commands. Thanks for your assistance, @ElMassimo!

**Comment by gbudiman** [comment]
I stumbled upon this thread while dealing with similar issue. It turns out my mistake is forgetting to turn _off_ `autoBuild` in my CI pipeline. (Yes I have `assets:precompile` in the preceding CI step)
